### PR TITLE
BKNDLSS-26899 stayLoggedIn fixed

### PR DESCRIPTION
**Pull request by holailusoria**

### CODE REVIEW

File: Backendless/Service/UserService.cs [hunk 1]
```diff
@@ -18,7 +18,32 @@ public class UserService
...
-    public BackendlessUser CurrentUser{ get; set; }
+    private BackendlessUser _currentUser = null;
+
+    public BackendlessUser CurrentUser
+    {
+      get
+      {
+        if( _currentUser == null )
+        {
+          if(LoginStorage.HasData)
```
Comment (markpiller) [inline Backendless/Service/UserService.cs:29]: @holailusoria can "LoginStorage" be null?

File: Backendless/Service/UserService.cs [hunk 2]
```diff
@@ -18,7 +18,32 @@ public class UserService
...
+          {
+            try
+            {
+              CurrentUser = Backendless.Data.Of<BackendlessUser>().FindById( LoginStorage.ObjectId );
```
Comment (markpiller) [inline Backendless/Service/UserService.cs:33]: Can ObjectId be null at this point?

**Review by markpiller — CHANGES_REQUESTED** [review]

File: Backendless/Service/UserService.cs (hunk 1 above)
Comment (holailusoria) [reply Backendless/Service/UserService.cs:29]: @markpiller line 29 checks for the presence of data in the LoginStorage.

File: Backendless/Service/UserService.cs (hunk 2 above)
Comment (holailusoria) [reply Backendless/Service/UserService.cs:33]: Line 29 checks for the presence of data in the LoginStorage.

File: Backendless/Service/UserService.cs (hunk 1 above)
Comment (holailusoria) [reply Backendless/Service/UserService.cs:29]: @markpiller no, it cant. An empty object is created in the LoginStorage getter if it is null. So it always returns a non-null LoginStorage object

**Pull request merged**